nistats/reporting.py

```diff
@@ -6,15 +6,170 @@
 
 import os
 import warnings
+from string import ascii_lowercase
+
 import numpy as np
+import pandas as pd
 from scipy import stats
 import nilearn.plotting  # overrides the backend on headless servers
+from nilearn.image.resampling import coord_transform
 import matplotlib
 import matplotlib.pyplot as plt
+from scipy import ndimage
 from patsy import DesignInfo
+
 from .design_matrix import check_design_matrix
 
 
+def _local_max(data, min_distance):
+    """Find all local maxima of the array, separated by at least min_distance.
+    Adapted from https://stackoverflow.com/a/22631583/2589328
+
+    Parameters
+    ----------
+    data : array_like
+        3D array of with masked values for cluster.
+
+    min_distance : :obj:`int`
+        Minimum distance between local maxima in ``data``, in terms of
+        voxels (not mm).
+
+    Returns
+    -------
+    ijk : :obj:`numpy.ndarray`
+        (n_foci, 3) array of local maxima indices for cluster.
+
+    vals : :obj:`numpy.ndarray`
+        (n_foci,) array of values from data at ijk.
+    """
+    data_max = ndimage.filters.maximum_filter(data, min_distance)
+    maxima = (data == data_max)
+    data_min = ndimage.filters.minimum_filter(data, min_distance)
+    diff = ((data_max - data_min) > 0)
+    maxima[diff == 0] = 0
+
+    labeled, num_objects = ndimage.label(maxima)
+    ijk = np.array(ndimage.center_of_mass(data, labeled, range(1, num_objects+1)))
+    ijk = np.round(ijk).astype(int)
+
+    vals = np.apply_along_axis(arr=ijk, axis=1, func1d=_get_val, input_arr=data)
```

Comment:
That's what it's meant to do, but for each row in ijk since ijk is 2d. I didn't think there was a better way of doing this, although it's always struck me as a little odd.